Sell/Basket/Tests/BasketRepositoryQueryTests.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  BasketRepositoryQueryTests.swift
+//  RealifeTechTests
+//
+//  Created by Mickey Lee on 25/06/2021.
+//  Copyright © 2021 Realife Tech. All rights reserved.
+//
+
+import XCTest
+import Apollo
+@testable import RealifeTech
+
+final class BasketRepositoryQueryTests: XCTestCase {
+
+    func test_getMyBasket_graphQLManagerHasData_completeWithSuccessCase() {
+        let (graphQLManager, sut) = BasketRepositoryTestHelper
+            .makeGraphQLManagerAndSUT(ofType: ApolloType.GetMyBasketQuery.Data.self)
+
+        let expectedResult = makeGetMyBasketGraphQLResult()
+        graphQLManager.resultReturns = .success(expectedResult)
+
+        let expectation = XCTestExpectation(description: "callback is fulfilled")
+        sut.getMyBasket { result in
+            guard case let .success(returnedResponse) = result else {
+                return XCTFail("This test should return success")
+            }
+            XCTAssertEqual(
+                returnedResponse.netAmount,
+                expectedResult.data?.getMyBasket?.fragments.fragmentBasket.netAmount)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.01)
+    }
+
+    func test_getMyBasket_returnsApolloClientError_completeWithFailureCase() {
+        let (graphQLManager, sut) = BasketRepositoryTestHelper
+            .makeGraphQLManagerAndSUT(ofType: ApolloType.GetMyBasketQuery.Data.self)
+
+        graphQLManager.resultReturns = .failure(DummyError.failure)
+
+        let expectation = XCTestExpectation(description: "callback is fulfilled")
+        sut.getMyBasket { result in
+            guard
+                case let .failure(returnedError) = result,
+                case let .regularError(error) = returnedError
+            else {
+                return XCTFail("This test should return failure")
+            }
+            XCTAssertEqual(error as? DummyError, DummyError.failure)
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 0.01)
+    }
+
+    func test_transformErrorForEveryQueryAndMutation_returnsCustomError_completeWithTransformedError() {
+        let errors: [BasketError] = [
+            .outOfStock,
+            .priceChange,
+            .mixedBasket,
+            .emptyBasket,
+            .regularError(GraphQLManagerError.noDataError)
+        ]
```

Comment:
I saw this BasketError list also be used in `BasketRepositoryMutationTests`, what's your thought moving it to `BasketRepositoryTestHelper`?